aiocache/base.py

```diff
@@ -109,18 +110,26 @@ class BaseCache:
     NAME: str
 
     def __init__(
-        self, serializer=None, plugins=None, namespace=None, key_builder=None, timeout=5, ttl=None
+        self,
+        serializer: Optional[BaseSerializer] = None,
+        plugins: Optional[List[Any]] = None,  # aiocache.plugins depends on aiocache.base
+        namespace: str = "",
+        key_builder: Optional[Callable[[str, str], str]] = None,
+        timeout: Optional[float] = 5,
+        ttl: Optional[float] = None,
     ):
-        self.timeout = float(timeout) if timeout is not None else timeout
-        self.namespace = namespace
-        self.ttl = float(ttl) if ttl is not None else ttl
-        self.build_key = key_builder or self._build_key
+        self.timeout: Optional[float] = float(timeout) if timeout is not None else None
+        self.ttl: Optional[float] = float(ttl) if ttl is not None else None
 
-        self._serializer = None
-        self.serializer = serializer or serializers.StringSerializer()
+        self.namespace: str = namespace
+        self._build_key: Callable[[str, str], str] = key_builder or (
+            lambda key, namespace="": f"{namespace}{key}")
 
-        self._plugins = None
-        self.plugins = plugins or []
+        self._serializer: Optional[BaseSerializer] = None
+        self.serializer: BaseSerializer = serializer or StringSerializer()
+
+        self._plugins: List[Any] = None
+        self.plugins: List[Any] = plugins or []
```

Comment:
```suggestion
        self._serializer = None
        self.serializer = serializer or StringSerializer()

        self._plugins: Optional[List[BasePluign]] = None
        self.plugins = plugins or []
```

Not sure what I'm missing, but `_plugins` doesn't make much sense. I'm not sure why it's initialised to `None` or where it later gets set to something else.

Reply:
`plugins` is also a property method, so that's broken too... Maybe someone meant to do `_plugins = plugins or []` and then use the property method?

Reply:
You're right.  This is strange.  I'll reduce these two initialization assignments into one, for `_serializer` and for `_plugins`.

`self.serializer` and `self._serializer` are used interchangeably throughout this module.  `self.serializer` would be appropriate in all cases.

---
Although the getter/setter don't do anything useful now, they do provide a mechanism for subclasses to do any customizations they might need.  I'd like to defer removing them to a future PR.

Reply:
Will leave any getter/setter changes to a subsequent PR.